### PR TITLE
fix(graphql): correct searchable instance types

### PR DESCRIPTION
**Pull request by sundersc**
- This change adds the OpenSearch instance types allowed by cloudformation.
- This reverts the incorrect instance types added in PR https://github.com/aws-amplify/amplify-cli/pull/9834.
- This is a short term solution to unblock the customers.
- Long term solution is tracked in https://github.com/aws-amplify/amplify-cli/issues/10075

Fixes https://github.com/aws-amplify/amplify-cli/issues/9956

### CODE REVIEW

**Comment by wvidana** [comment]
Why not just remove the `allowedValues` altogether? Just let CFN define those. That way no one has to maintain the list and people that need to run an older version of the CLI are not stuck with an old list.

**Comment by sundersc** [comment]
If we remove the allowed values list and someone misspell an instance type, the error would appear at much later time on push resulting in poor DX. We are thinking about using SDK to retrieve the allowed instance types.
https://docs.aws.amazon.com/opensearch-service/latest/developerguide/configuration-api.html#configuration-api-actions-listinstancetypedetails

**Review by alharris-at — COMMENTED** [review]
One q, otherwise LGTM

**Review by marcvberg — APPROVED** [review]
LGTM

File: packages/amplify-graphql-searchable-transformer/src/cdk/create-cfnParameters.ts [hunk 1]
```diff
@@ -157,85 +157,45 @@ export function createParametersStack(stack: Stack): Map<string, CfnParameter> {
           'c6g.4xlarge.elasticsearch',
           'c6g.xlarge.elasticsearch',
           'c6g.12xlarge.elasticsearch',
-          't3.small.search',
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-graphql-searchable-transformer/src/cdk/create-cfnParameters.ts:160]: The list is maintained separately for v1 and v2 transformers. It was intentionally maintained in different places to keep v1 and v2 package independent of each other. The V1 will eventually go away.

**Review by alharris-at — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]
LGTM!

**Comment by ataibarkai** [comment]
@sundersc why were all the `X.Y.search` instance types removed, and only the `X.Y.elasticsearch` instance types kept?

I believe `.elasticsearch` instance types are deprecated/outdated, and `.search` instances are now the preferred ones. E.g. see [this document](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/rename.html#rename-instances) describing the OpenSearch summary of changes that occurred on September 8th 2021:
> Instance types in Amazon OpenSearch Service are now in the format `<type>.<size>.search`—for example, `m6g.large.search` rather than `m6g.large.elasticsearch`

That document goes on to say that 
> "Existing domains will start automatically referring to the new instance types within the API and in the Billing and Cost Management console" 

BUT it seems that if this is going to happen, it hasn't yet. `.elasticsearch` instance types appear to result in outdated deployments which behave different from the latest ones, and which the AWS OpenSearch console prompts you to upgrade.

**Pull request merged**